react/features/recording/components/RecordingLabel.web.js

```diff
@@ -0,0 +1,303 @@
+        super(props);
+
+        this.state = {
+            /**
+             * Whether or not the filmstrip was not visible but has transitioned
+             * in the latest component update to visible. This boolean is used
+             * to set a class for position animations.
+             *
+             * @type {boolean}
+             */
+            filmstripBecomingVisible: false,
+
+            /**
+             * A cache of the _recordingState prop. It is compared to the next
+             * value of _recordingState on component update to identify from
+             * which recording state a failure was encountered.
+             *
+             * @type {string|null}
+             */
+            previousRecordingState: null
+        };
+    }
+
+    /**
+     * Updates the state for whether or not the filmstrip is being toggled to
+     * display after having being hidden and caches the previous recording
+     * state.
+     *
+     * @inheritdoc
+     * @param {Object} nextProps - The read-only props which this Component will
+     * receive.
+     * @returns {void}
+     */
+    componentWillReceiveProps(nextProps) {
+        this.setState({
+            previousRecordingState: this.props._recordingState,
+            filmstripBecomingVisible: nextProps._filmstripVisible
+                && !this.props._filmstripVisible
+        });
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { centered, key, showSpinner } = this._getDisplayConfiguration();
+        const isVisible = this.props._shouldDisplay && Boolean(key);
+        const rootClassName = [
+            'video-state-indicator centeredVideoLabel',
+            isVisible ? 'show-inline' : '',
+            centered ? '' : 'moveToCorner',
+            this.state.filmstripBecomingVisible ? 'opening' : '',
+            this.props._filmstripVisible
+                ? 'with-filmstrip' : 'without-filmstrip',
+            this.props._remoteVideosVisible
+                ? 'with-remote-videos' : 'without-remote-videos'
+        ].join(' ');
+
+        return (
+            <span
+                className = { rootClassName }
+                id = 'recordingLabel'>
+                <span id = 'recordingLabelText'>
+                    { this.props.t(key) }
+                </span>
+                { showSpinner
+                    ? <img
+                        className = 'recordingSpinner'
+                        id = 'recordingSpinner'
+                        src = 'images/spin.svg' />
+                    : null }
+            </span>
+        );
+    }
+
+    /**
+     * Creates a configuration object describing how @{code RecordingLabel}
+     * should be displayed -- the text content, loading spinner display, and
+     * centered in parent.
+     *
+     * @private
+     * @returns {Object}
+     */
+    _getDisplayConfiguration() {
```

Comment:
It looks like this function duplicates all checks from Recording.js->updateRecordingUI. The only other thing that's done in the updateRecordingUI is to _setToolbarButtonToggled state. I think we can already do that in react with toggleToolbarButton and get rid of the direct css change we're currently doing. This then raise a question if the right place for the getDisplayConfiguration is in the RecordingLabel component itself or should it be separated, as it will operate over multiple components.